tests/system/src/test/java/test/com/sun/marlin/HugePolygonClipTest.java

```diff
@@ -0,0 +1,284 @@
+import javax.imageio.ImageIO;
+import javax.imageio.ImageWriteParam;
+import javax.imageio.ImageWriter;
+import javax.imageio.stream.ImageOutputStream;
+
+import junit.framework.AssertionFailedError;
+import org.junit.AfterClass;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import test.util.Util;
+import static test.util.Util.TIMEOUT;
+
+/**
+ * Clip rendering test
+ *
+ * @test
+ * @summary verify that huge polygon is properly rasterized
+ * @bug 8274066
+ */
+public class HugePolygonClipTest {
+
+    private static final Double LARGE_X_COORDINATE = 4194304.250;
+    private static final int SCENE_WIDTH = 600;
+    private static final int SCENE_HEIGHT = 400;
+
+    private static final double WIDTH = 2.73;
+
+    private static final int G_MASK = 0x0000ff00;
+    private static final int R_MASK = 0x00ff0000;
+    private static final int RGB_MASK = 0x00ffffff;
+
+    private static final boolean SAVE_IMAGE = false;
+
+    // Used to launch the application before running any test
+    private static final CountDownLatch launchLatch = new CountDownLatch(1);
+
+    static {
+        Locale.setDefault(Locale.US);
+        /*
+            System.out.println("BLUE_PIXEL: " + BLUE_PIXEL);
+            System.out.println("RED_PIXEL:  " + RED_PIXEL);
+         */
+        // enable Marlin logging:
+        System.setProperty("prism.marlin.log", "true");
+
+        System.setProperty("prism.marlin.clip", "true");
+
+        System.setProperty("prism.marlin.subPixel_log2_X", "8");
+    }
+
+    // Singleton Application instance
+    static MyApp myApp;
+
+    // Application class. An instance is created and initialized before running
+    // the first test, and it lives through the execution of all tests.
+    public static class MyApp extends Application {
+
+        Stage stage = null;
+
+        public MyApp() {
+            super();
+        }
+
+        @Override
+        public void init() {
+            HugePolygonClipTest.myApp = this;
+        }
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            this.stage = primaryStage;
+
+            stage.setScene(new Scene(new Group()));
+            stage.setTitle("HugePolygonClipTest");
+            stage.show();
+
+            launchLatch.countDown();
+        }
+    }
+
+    @BeforeClass
+    public static void setupOnce() {
+
+        // Start the Application
+        new Thread(() -> Application.launch(MyApp.class, (String[]) null)).start();
+
+        try {
+            if (!launchLatch.await(TIMEOUT, TimeUnit.MILLISECONDS)) {
+                throw new AssertionFailedError("Timeout waiting for Application to launch");
+            }
+
+        } catch (InterruptedException ex) {
+            AssertionFailedError err = new AssertionFailedError("Unexpected exception");
+            err.initCause(ex);
+            throw err;
+        }
```

Comment:
Suggestion: if you add `throws Exception` to the method signature, this try/catch block can be simplified to a single assert statement:

```
    assertTrue("Timeout waiting for Application to launch",
            launchLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
```

Many of our older tests use the patter in this method, but most newer tests use the simplified check.